### ISSUE TITLE
devtools shutdown remote production database when unit test done

### ISSUE THREAD

**Issue by SolaKun** [spring-projects/spring-boot#5753]
I'm using spring-boot-devtools for template auto-reloading propose, but lately I found that the dev-tools will shutdown my remote production database strangely after every spring-jpa integration unit test was done .
I searched about devtools in spring-boot docs, but nothing referred to this issue.

Database OS: Centos 7
Database: Mariadb-10.1.13

Shutdown log:
```
[Note] /usr/sbin/mysqld: Normal shutdown
[Note] Event Scheduler: Purging the queue. 0 events
[Note] InnoDB: FTS optimize thread exiting.
[Note] InnoDB: Starting shutdown...
[Note] InnoDB: Shutdown completed; log sequence number 2069021
[Note] /usr/sbin/mysqld: Shutdown complete
```

**Comment by wilkinsona** [comment]
What version of Boot are you using and how have you configured the DataSource?

**Comment by SolaKun** [comment]
Spring-Boot: 1.3.3.RELEASE
Using C3P0's ComboPooledDataSource as DataSource

Bean configuration:
``` java
	@Bean
	DataSource dataSource(
			@Value("${c3p0.config.name}") String name) {
		if (name.equalsIgnoreCase("default")) {
			return new ComboPooledDataSource();
		} else {
			return new ComboPooledDataSource(name);
		}
	}
```

c3p0-config.xml
``` xml
<c3p0-config>
	<default-config>
		<property name="driverClass">org.hsqldb.jdbcDriver</property>
		<property name="maxPoolSize">10</property>
		<property name="minPoolSize">2</property>
		<property name="acquireIncrement">1</property>
		<property name="jdbcUrl">jdbc:hsqldb:mem:test</property>
	</default-config>
	<named-config name="remote-test" >
		<property name="driverClass">com.mysql.jdbc.Driver</property>
		<property name="maxPoolSize">10</property>
		<property name="minPoolSize">2</property>
		<property name="acquireIncrement">1</property>
		<property name="user">root</property>
		<property name="password">******</property>
		<property name="jdbcUrl">
			jdbc:mysql://myhost:3306/mydatabase?autoReconnect=true&amp;characterEncoding=utf8
		</property>
		<property name="idleConnectionTestPeriod">1800</property>
		<property name="maxIdleTime">3600</property>
	</named-config>
</c3p0-config>
```

test case
``` java
@RunWith(SpringJUnit4ClassRunner.class)
@SpringApplicationConfiguration(TestApplication.class)
@TestPropertySource(properties = "spring.jpa.show-sql=true")
@TestPropertySource(properties = "c3p0.config.name=remote-test")
public class AbstractRemoteJPAIntegrationTestextends AbstractJUnit4SpringContextTests {
}
```

**Comment by wilkinsona** [comment]
Thanks. DevTools is getting confused because you have both an in-memory database driver and the MySQL driver on the classpath and you're not using Boot's DataSource auto-configuration. This has been fixed by the changes made for #5540.
